### PR TITLE
Location Layer Android bindings

### PR DESCRIPTION
**Pull request by LukasPaczos**
This PR adds Android bindings for the location layer and fixes a lost precision issue by adding an `array<double>` converter.

This reuses the `PropertyValue` converter implementation from the `location_component_layer.cpp`, but it feels to me like it should live in a different place, I just couldn't find where.

Please take this branch over @galinelle rebase/merge or fixup anything that needs to be done.

/cc @tmpsantos @alexshalamov

### CODE REVIEW

**Comment by alexshalamov** [comment]
@galinelle could you pick non-android related changes to your PR? Thanks.

**Comment by LukasPaczos** [comment]
@alexshalamov I noticed that @galinelle already did it, so I rebased and cleaned up this PR.

File: platform/android/src/style/layers/background_layer.cpp [hunk 1]
```diff
@@ -26,13 +26,16 @@ namespace android {
     /**
      * Creates a non-owning peer object (for layers currently attached to the map)
      */
-    BackgroundLayer::BackgroundLayer(mbgl::style::BackgroundLayer& coreLayer) : Layer(coreLayer) {}
+    BackgroundLayer::BackgroundLayer(mbgl::style::BackgroundLayer& coreLayer)
```
Comment (LukasPaczos) [inline platform/android/src/style/layers/background_layer.cpp:29]: Looks like the generated file changes have not been committed in the past, have we disabled the checks @tmpsantos?
The `.ejs` template file is identical to the version in `galinelle_locationComponent` and `master`.

**Comment by LukasPaczos** [comment]
This is ready for review, but we can also wait for https://github.com/mapbox/mapbox-gl-native/pull/16340/ to land and I can retarget to `master`.

**Comment by LukasPaczos** [comment]
Closing, since platform bindings have been moved to https://github.com/mapbox/mapbox-gl-native-android. Once https://github.com/mapbox/mapbox-gl-native/pull/16340/ lands, I'll redo this PR in the platform repo.

**Pull request closed without merging**